### DISCUSSION TITLE
Pagination crashes when data is updated

### DISCUSSION THREAD

**Discussion by iakovalgaev** [table-library/react-table-library#52] [General]
Let's say I'm on the fifth page. I do some action after which I request updated data. When they come, it throws me from the fifth to the first page. I would like to be in control of this moment.

**Comment by iakovalgaev** [comment]
I'm sorry. I hurried and did not notice one moment. Tables thus retain page position by default.

**Reply by rwieruch** [reply]
Yes, the page keeps staying at the position. If new data comes in and you want to do something about the page, you can use the imperative API. Something like:

```
const pagination = usePagination(...);

React.useEffect(() => 
  if (data !== previousData) {
    pagination.fns.onSetPage(0);
  }
, [data]};
```
